app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryBadge.kt

```diff
@@ -1,55 +1,61 @@
 package eu.kanade.tachiyomi.ui.library
 
 import android.content.Context
-import android.graphics.Color
 import android.util.AttributeSet
 import android.view.View
-import androidx.core.content.ContextCompat
 import com.google.android.material.card.MaterialCardView
 import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.util.system.contextCompatColor
 import eu.kanade.tachiyomi.util.system.dpToPx
-import eu.kanade.tachiyomi.util.system.getResourceColor
+import eu.kanade.tachiyomi.util.view.gone
+import eu.kanade.tachiyomi.util.view.isNotGone
+import eu.kanade.tachiyomi.util.view.isVisible
 import eu.kanade.tachiyomi.util.view.updatePaddingRelative
+import eu.kanade.tachiyomi.util.view.visibleIf
 import kotlinx.android.synthetic.main.unread_download_badge.view.*
 
 class LibraryBadge @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) :
     MaterialCardView(context, attrs) {
 
     fun setUnreadDownload(unread: Int, downloads: Int, showTotalChapters: Boolean) {
         // Update the unread count and its visibility.
+
+        val unreadBadgeBackground = when (showTotalChapters) {
+            true -> context.contextCompatColor(R.color.total_badge)
+            false -> context.contextCompatColor(R.color.unread_badge)
+        }
+
         with(unread_text) {
+            visibleIf(unread > 0 || unread == -1 || showTotalChapters)
+
             text = if (unread == -1) "0" else unread.toString()
-            setTextColor(if (unread == -1 && !showTotalChapters)
-                context.getResourceColor(android.R.attr.colorAccent)
-            else Color.WHITE)
-            setBackgroundColor(
-                if (showTotalChapters) ContextCompat.getColor(context, R.color.material_deep_purple_500)
-                else context.getResourceColor(android.R.attr.colorAccent))
-            visibility = when {
-                unread > 0 || unread == -1 || showTotalChapters -> View.VISIBLE
-                else -> View.GONE
```

Comment:
had 1 question with all the elements here. If they are View.Gone is there a reason you are still setting all the fields?